src/utils/latency_tracer.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#pragma once
+#include <dsn/utility/synchronize.h>
+
+namespace dsn {
+namespace utils {
+
+/**
+ * latency_tracer is a simple tool for tracking request time consuming in different stages, which
+ * can help user to find out the latency bottleneck. user needs to use it to "add_point" in one
+ *stage,
+ * which will record the name of point and the timestamp. when the request is finshed, you can dump
+ * the formated result.
+ *
+ * for example: one request experiences four stages, latency_tracer need be held by request and
+ * passes all stages:
+ * class request {
```

Comment:
```suggestion
 * latency_tracer is a tool for tracking the time spent in each of the stages during request execution.
 * It can help users to figure out where the latency bottleneck is located.
 * User needs to use `add_point` before entering one stage, which will record the name
 * of this stage and its start time.
 * When the request is finished, you can dump the formatted result by `dump_trace_points`.
 *
 * For example, given a request with a 4-stage pipeline (the `latency_tracer` need to
 * be held by this request throughout the execution):
 *
 * ```
 * class request {
```